solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -2284,8 +2285,10 @@ public void run() {
 
       SolrCore core;
       while (!container.isShutDown() && (core = solrCores.getCoreToClose()) != null) {
+        assert core.getOpenCount() == 1;
```

Comment:
assert is only enabled in tests usually, right?

What happens if the core gets reopened while we close it here? Is it a case of the open count being higher than one in a legitimate way?

Reply:
The closer thread only operates on cores that have already been removed from access (e.g. to be opened for a new query).  And a request to open it (transiently?) will block because this queue is considered in `SolrCores.pendingCoreOps`

Reply:
> assert is only enabled in tests usually, right?

Right.